src/helpers/format.ts

```diff
@@ -1,10 +1,53 @@
 import type { ReactTestRendererJSON } from 'react-test-renderer';
-import prettyFormat, { plugins } from 'pretty-format';
+import prettyFormat, { NewPlugin, plugins } from 'pretty-format';
 
-const format = (input: ReactTestRendererJSON | ReactTestRendererJSON[]) =>
+type MapPropsFunction = (
+  props: Record<string, unknown>,
+  node: ReactTestRendererJSON
+) => Record<string, unknown>;
+
+export type FormatOptions = {
+  mapProps?: MapPropsFunction;
+};
+
+const format = (
+  input: ReactTestRendererJSON | ReactTestRendererJSON[],
+  options: FormatOptions = {}
+) =>
   prettyFormat(input, {
-    plugins: [plugins.ReactTestComponent, plugins.ReactElement],
+    plugins: [getCustomPlugin(options.mapProps)],
     highlight: true,
   });
 
+const getCustomPlugin = (mapProps?: MapPropsFunction): NewPlugin => {
+  return {
+    test: (val) =>
+      plugins.ReactTestComponent.test(val) || plugins.ReactElement.test(val),
+    serialize: (val, config, indentation, depth, refs, printer) => {
+      if (plugins.ReactTestComponent.test(val)) {
+        if (mapProps && val.props) {
+          val.props = mapProps(val.props, val);
```

Comment:
Not sure if it's a good idea to mutate the passed `val` object. Can we rather use object spread operator to create a modified copy?

Reply:
I had my doubt on this, this is why I added a test to verify that the original object is not modified and it's not the case so I think it's fine as is. Also I believe that using the spread operator would create a shallow copy so it would not really be a protection against modifications either

Reply:
format works on JSON output not ReactTestInstances, but still invoking `format` would mutate the JSON output. I've changed that to object copy, not mutation.